lib/components/narrative/narrative-itineraries.js

```diff
@@ -145,8 +147,8 @@ class NarrativeItineraries extends Component {
                 onClick={this._toggleDetailedItinerary}>
                 <i className='fa fa-arrow-left' /> View all options
               </button>
-              {/* FIXME: only save if meets requirements (e.g., is transit + non-realtime mode) */}
-              {persistence && persistence.enabled
+              {/* Only save if itinerary has transit and no rental modes that cannot be guaranteed. */}
+              {persistence && persistence.enabled && itineraryCanBeMonitored(itineraries[activeItinerary])
```

Comment:
So, I think we should maybe take this opportunity to encapsulate this button into its own component. There are a few states that could be represented here, that we haven't quite addressed yet:
1. User is not logged in - in this instance, I think we should have the SaveTripButton render something like: "Please sign in to save trip"
2. Persistence disabled - just return null
3. itin cannot be monitored - we should probably disable the button and for now show a tooltip saying: "To save a trip it must contain a transit leg and no rental vehicles legs." Not sure if that's a great message long-term though...

Reply:
1. I agree it is probably the simpler solution for now.
2. Agree
3. How about for the disabled button text: `🚫 Cannot save`? And for the tooltip, taken from the backend: "Only trips that include transit and no rentals or ride hailing can be saved/monitored."